mmv1/products/pubsub/terraform.yaml

```diff
@@ -142,8 +152,18 @@ overrides: !ruby/object:Overrides::ResourceOverrides
       encoder: templates/terraform/encoders/no_send_name.go.erb
       extra_schema_entry: templates/terraform/extra_schema_entry/pubsub_subscription_path.erb
       update_encoder: templates/terraform/update_encoder/pubsub_subscription.erb
-
-
+  Schema: !ruby/object:Overrides::Terraform::ResourceOverride
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "pubsub_schema_basic"
+        primary_resource_id: "example"
+        vars:
+          schema_name: "example"
+    properties:
+      name: !ruby/object:Overrides::Terraform::PropertyOverride
+        diff_suppress_func: 'compareSelfLinkOrResourceName'
+        custom_expand: templates/terraform/custom_expand/resource_from_self_link.go.erb
+        custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
```

Comment:
So the tests are failing because there is a persistent diff after apply. This is most likely because the definition field is not returned during read so terraform believes there is a diff. You can test this locally by applying the resource and running a subsequent plan.

```
------- Stdout: -------
=== RUN   TestAccPubsubSchema_pubsubSchemaBasicExample
=== PAUSE TestAccPubsubSchema_pubsubSchemaBasicExample
=== CONT  TestAccPubsubSchema_pubsubSchemaBasicExample
    provider_test.go:275: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_pubsub_schema.example will be updated in-place
          ~ resource "google_pubsub_schema" "example" {
              + definition = jsonencode(
                    {
                      + fields = [
                          + {
                              + name = "StringField"
                              + type = "string"
                            },
                          + {
                              + name = "IntField"
                              + type = "int"
                            },
                        ]
                      + name   = "Avro"
                      + type   = "record"
                    }
                )
                id         = "projects/ci-test-project-188019/schemas/example643x21yod0"
                name       = "example643x21yod0"
                # (2 unchanged attributes hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccPubsubSchema_pubsubSchemaBasicExample (6.82s)
FAIL

```

You can fix this by adding [ignore read](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/products/accesscontextmanager/terraform.yaml#L71) to the definition field. Unfortunately this means if this field is returned it won't be measured against in an update. To customize reads and compares on json fields [check out this field](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/products/bigquery/terraform.yaml#L380-L385).